### PR TITLE
[OTS-338] Remove the need for a fixed size container

### PR DESCRIPTION
**Pull request by AntonioMA**
r? @adrice727 
cc @marinaserranomontes 
This makes the annotations work on a container that does not have the size fixed to the video size (should work in any container in opentokdemo for example). The way it works is:
- The video dimension is fixed independently of the container.
- Calculate the shown video dimension taking into account that the video will cover either the whole width or the whole height of the container (depending on the ratio)
- Calculate the space where the video is shown
- Put the canvas on the right place
- Use coordinates relative to the actual video size instead of the canvas size

I've only tested this on a canvas that's inside a window. And I've only tested this between two browsers. The good news is that it works well even on wormhole shared screens.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/opentok/annotation-acc-pack/pull/62" target="_blank">ReviewNinja

**Comment by albertoacn** [comment]
:+1 for ninja-review

**Comment by adrice727** [comment]
@AntonioMA These changes break annotation in the external window.  I haven't had time to dig into why yet.

**Comment by AntonioMA** [comment]
@adrice727 Probably because I didn't fix the external window branch in draw...

**Comment by AntonioMA** [comment]
Is there some easy way for me to test that? (as in a easy to use sample that opens on a external window that I can use to fix the other branch?)

File: js/opentok.js-annotations/src/acc-pack-annotation.js [hunk 1]
```diff
@@ -139,16 +139,8 @@
...
-
-      if (computedHeight <= windowDimensions.height) {
-        width = windowDimensions.width;
-        height = computedHeight;
-      } else {
-        height = windowDimensions.height;
-        width = height * _aspectRatio;
-      }
+      width = windowDimensions.width;
+      height = windowDimensions.height;
```
Comment (AntonioMA) [inline js/opentok.js-annotations/src/acc-pack-annotation.js]: I think this is the only change needed for it to work on a external window. I haven't been able to test it though (dunno how to open it on a external window!).

Comment (AntonioMA) [inline js/opentok.js-annotations/src/acc-pack-annotation.js]: @adrice727 Please test it again :)

**Comment by adrice727** [comment]
@AntonioMA Just to be sure, did you run the build process?  I'm using the `dist` file to test and the alignment of annotations is off between publisher and subscriber.

**Comment by AntonioMA** [comment]
Sorry for the delay in answering. I've run the build.sh -d script. The result of that is at 
https://opentokdemo.tokbox.com/js/vendor/opentok-annotation.js and it's being used by opentokdemo if you pass the enableAnnotations parameter.

Attached are two images, of a publisher and a subscriber annotated using that on opentokdemo.

There is a known issue (well, known to me anyway 😄 ) that is *not* a problem of this code. This code (and the old one) relies on knowing the size of the canvas container and being notified when that changes. But apparently some CSS transformations do not fire the onResize event. That's why you sometimes have to manually resize the window a little. So if you're getting alignment problems, try resizing the window manually once the shared screen is shown, and before you annotate. 

screen shot 2016-09-20 at 14 55 02
screen shot 2016-09-20 at 14 55 17

**Comment by adrice727** [comment]
@AntonioMA There are some merge conflicts due to some changes that were merged last week.  Is it easier for you to pull/merge upstream and update the PR, or would you like me to manually commit the changes?

**Comment by adrice727** [comment]
@AntonioMA Alignment is still off of when using the external window.

**Comment by AntonioMA** [comment]
Oh, BTW, I can promise and will promise that I haven't touched anything on iOS 😛

**Comment by adrice727** [comment]
👍

**Pull request merged**